File: tests/class/weak.js
export function allocate() {
  const weakSelf = new WeakSelf();
  weakSelf.property = null;
  const result = weakSelf;
  result.property = result;
  return result;
}
export function read$weakSelf$(weakSelf) {
  return weakSelf.property;
}
export class WeakSelf {}

File: tests/class/weak.swift
public class WeakSelf {
	weak var property: WeakSelf?
	init() {
	}
}

public func allocate() -> WeakSelf {
	let result = WeakSelf()
	result.property = result
	return result
}

public func read(weakSelf: WeakSelf) -> WeakSelf? {
	return weakSelf.property
}
